ksmt-core/src/main/kotlin/org/ksmt/decl/KBitVecExprsDecl.kt

```diff
@@ -11,23 +12,51 @@ import org.ksmt.sort.KBvSort
 import org.ksmt.sort.KBoolSort
 import org.ksmt.sort.KBv1Sort
 import org.ksmt.sort.KIntSort
-import org.ksmt.utils.toBinary
 import java.math.BigInteger
 
 abstract class KBitVecValueDecl<T : KBvSort> internal constructor(
     ctx: KContext,
-    val value: String,
+    val value: Number,
     sort: T
-) : KConstDecl<T>(ctx, "#b$value", sort) {
-    internal constructor(ctx: KContext, value: Number, sort: T) : this(ctx, value.toBinary(), sort)
+) : KConstDecl<T>(ctx, mkDeclName(ctx, value, sort.sizeBits), sort) {
+    companion object {
+        private fun mkDeclName(ctx: KContext, value: Number, sizeBits: UInt): String =
+            if (ctx.printerParams.bvValuePrintMode == BvValuePrintMode.HEX && sizeBits % HALF_BYTE == 0u) {
+                "#x${value.toHexString(sizeBits / HALF_BYTE)}"
```

Comment:
I didn't get, why is it called with `sizeBits / HALF_BYTE`. Why an integer should be matched as a byte value? Please, add a comment about it.